### ISSUE TITLE
Assemble Atmel Studio parts and build

### ISSUE THREAD

**Issue by HakanL** [spincraft/IoT-WINC1500#16]
This task is to build the solution based on the demo1 project, but connected to Azure IoT instead of PubNub.

**Comment by HakanL** [comment]
I just pushed some more code. I've added RTCZero which seems like it's needed. I've put some of the "adapters" back in and renamed some from arduino to samd21. I'm currently getting 11 build errors, but I'm sure that's just the beginning as more will most likely pop up when these are fixed. Here's a short list of the issues:

```
		src/Azure-iot-sdk/adapters/tlsio_samd21.o: In function `tlsio_arduino_create':
C:\Projects\Electronics\IoT-WINC1500\WINC1500_IOT_CLOUD_DEMO1\WINC1500_IOT_CLOUD_DEMO1\src\Azure-iot-sdk\adapters\tlsio_samd21.c(168,1): error: undefined reference to `sslClient_setTimeout'
C:\Projects\Electronics\IoT-WINC1500\WINC1500_IOT_CLOUD_DEMO1\WINC1500_IOT_CLOUD_DEMO1\src\Azure-iot-sdk\adapters\tlsio_samd21.c(168,1): error: undefined reference to `sslClient_hostByName'
		src/Azure-iot-sdk/adapters/tlsio_samd21.o: In function `tlsio_arduino_send':
C:\Projects\Electronics\IoT-WINC1500\WINC1500_IOT_CLOUD_DEMO1\WINC1500_IOT_CLOUD_DEMO1\src\Azure-iot-sdk\adapters\tlsio_samd21.c(401,1): error: undefined reference to `sslClient_write'

```

At the moment I'm not sure how to fix these, the requested functions all live in sslClient_samd21.h/c, but I don't know why it doesn't compile. Any help would be appreciated.

**Comment by nikolxx** [comment]
The sslClient_samd21.h header file doesn't define the functions as I'm used to (I haven't worked with c++ so maybe that's part of my confusion?), and so maybe the compiler isn't picking up the functions from the header file. I'm not familiar with their use of:
`MOCKABLE_FUNCTION(, void, sslClient_setTimeout, unsigned long, timeout);`
which is apparently:
`#define MOCKABLE_FUNCTION(modifiers, result, function, ...) \
    MOCKABLE_FUNCTION_UMOCK_INTERNAL_WITH_MOCK(modifiers, result, function, __VA_ARGS__)`
Basically, I don't know either but maybe looking into this could yield something, unless someone else knows what this is.

**Comment by nikolxx** [comment]
Regarding RTCZero:

1. We may not need the Arduino RTCZero library. The Atmel RTC driver appears in the project already, src\ASF\sam0\drivers\rtc\, but I have not tried to use it. The API ref is [here](http://asf.atmel.com/docs/3.32.0/samd21/html/group__asfdoc__sam0__rtc__calendar__group.html), and there is example implementation code [here](http://asf.atmel.com/docs/3.32.0/samd21/html/asfdoc_sam0_rtc_calendar_exqsg.html). Once thing about the ASF drivers I've used is that they are all C instead of C++, so they may be understandable by more people.

2. Licensing: @spincraft we need some direction here - what licensing are we shooting for? See Issue #3  This Arduino library is LGPL so would require anyone using this project to release at least part of the code. The Azure IoT example is MIT so it can be integrated into proprietary products. If LGPL should not be in the roadmap, then we should avoid pulling in Arduino libraries.

**Comment by HakanL** [comment]
The current state is a mix of the cloud1 pubnub sample plus the azure-iot-arduino. Then I've started to customize it to try to make it work.

The MOCKABLE_FUNCTION is something that was there in the source already, it seems to just be a define, maybe I should just try to remove that part for now.

I'm not sure 100% sure we need RTCZero, I figured let's try to get it to run first and then we can cleanup :)

**Comment by HakanL** [comment]
I changed them to regular definitions, same issue though :( It seems that all the Azure SDK methods are defined with this MOCKABLE_FUNCTION so I don't think it's the cause of the issues.

**Comment by spincraft** [comment]
RE: licensing
excellent question - by choosing the MKR1000, we have an open source hardware component. Most of the software stack is open, as is my understanding. When this project is posted on Azure Marketplace, in Atmel Studio ASF and newly added, Arduino Community, it will have open source but limited # of channels(1 input{temperature}, 1 output{LED} & 1 user control{setpoint for temperature}). It should run as a functional demo and promote the SD Hackers as experts to customize a solution for the client. I see it as identical to National Inst. System Integrators designing a test solution. The Integrator gets paid to use the IoT template design and write routines for the various sensors, indicators and actuators required to build a complete solution. These solutions are proprietary to each client. The proprietary data gathered is also analysed, charted and reported. As part of this complete breakfast, Stacks test system provides test stim to the MKR1000 to perform functional testing on MKR based IoT nodes, as well as provide sample data to run data simulations to develop the cloud-based analysis and trending charts. Multi-node traffic can also be simulated. IoT Node battery life testing is essential prior to deployment, using the Stacks DMM functionality. Each client will be different but lack the resources and experience to create their custom solution.
my $0.02 worth
Don
